lib/pbench/test/unit/agent/test_tool_meister.py

```diff
@@ -0,0 +1,51 @@
+"""Tests for the Tool Meister module.
+"""
+
+import logging
+import os
+import pytest
+
+from pbench.agent.tool_meister import log_pipe_output
+from pbench.agent.tool_meister import Tool
+
+
+def test_log_pipe_output(caplog):
+    read_side, write_side = os.pipe()
+
+    with os.fdopen(write_side, "wb") as wfp:
+        wfp.write(b"line 1\n\nline3\nno-eol line 4")
+
+    with os.fdopen(read_side, "rb") as pipe:
+        log_pipe_output(pipe, logging.getLogger("test_log_pipe_output"))
+
+    assert caplog.record_tuples[0][2] == "line 1"
+    assert caplog.record_tuples[1][2] == "line3"
+    assert caplog.record_tuples[2][2] == "no-eol line 4"
```

Comment:
Rather than actually creating a pipe, consider making `pipe` just a simple object with a `readlines()` method which returns the value at line 16.